src/server/rpc/procedures/utils/api-consumer.js

```diff
@@ -0,0 +1,272 @@
+                msgContents.forEach(content=>{
+                    let msg = {
+                        dstId: this.socket.roleId,
+                        msgType,
+                        content
+                    };
+                    remainingMsgs[this.socket.uuid].push(msg);
+                });
+                this._logger.trace(`initializing sending of ${msgContents.length} messages`);
+                this._sendNext();
+            });
+    }
+
+    /**
+     * queries the endpoint for an specific string answer.
+     * @param  {object} queryOptions
+     * @param  {string} selector     json-query-like string
+     * @return {Promise}
+     */
+    _sendAnswer(queryOptions,selector){
+        return this._requestData(queryOptions)
+            .then(res => {
+                let answer = this._queryJson(res,selector);
+                this._logger.trace('answer is', answer);
+                this.response.send(answer);
+            });
+    }
+
+    // sends an image to the user
+    _sendImage(queryOptions){
+        let logger = this._logger,
+            response = this.response;
+        var imageBuffer = new Buffer(0);
+        let imgResponse = this._requestImage(queryOptions);
+        imgResponse.on('data', function(data) {
+            imageBuffer = Buffer.concat([imageBuffer, data]);
+        });
+
+        imgResponse.on('end', function() {
+            if (imgResponse.isImage){
+                response.set('cache-control', 'private, no-store, max-age=0');
+                response.set('content-type', 'image/png');
+                response.set('content-length', imageBuffer.length);
+                response.set('connection', 'close');
+                response.status(200).send(imageBuffer);
+                logger.trace('sent the image');
+            }
+        });
+        imgResponse.on('error', err => {
+            logger.error(err);
+            // QUESTION what to return in case of an error?
+            response.status(404).send('null');
+        });
+    }
+
+    // WIP needs further testing
+    _makeHelpers(argsArr,fnMap, queryOptionsMaker){
+        fnMap.forEach(fnEntry => {
+            this[fnEntry.name] = (...argsArr) => {
+                // QUESTION can I pass in an obj with place holder for future defined variables?
+                let queryOptions = queryOptionsMaker(...argsArr);
+                this._sendAnswer(queryOptions,fnEntry.selector);
+                return null;
+            };
+        });
+    }
+
+    // helper test the response
+    // = request data that logs then logs the response and limits it by selector
+    _inspectResponse(queryOptions, selector){
+        this._requestData(queryOptions)
+            .then(res=> {
+                this._logger.trace('got response');
+                this._logger.trace(this._queryJson(res,selector));
+            });
+    }
+
+    // QUESTION how to make this apply whenever sendMsgs() is used?
```

Comment:
I don't think this is an issue. we can probably remove this comment as long as it all works